gaphor/diagram/presentation.py

```diff
@@ -328,8 +328,10 @@ def postload(self):
             assert self._connections.get_connection(self.tail)
             del self._load_tail_connection
 
-    def _on_orthogonal(self, event):
-        self._set_orthogonal(event.new_value)
+    def _on_orthogonal(self, _event):
+        if self.orthogonal and len(self.handles()) < 3:
+            raise ValueError("Can't set orthogonal line with less than 3 handles")
+        self.update_orthogonal_constraints(self.orthogonal)
 
-    def _on_horizontal(self, event):
-        self._set_horizontal(event.new_value)
+    def _on_horizontal(self, _event):
+        self.update_orthogonal_constraints(self.orthogonal)
```

Comment:
Do we need to pass in that `horizontal=True`?

Reply:
No, since we only need to update the orthogonal constraints when the "orthogonal" property is True.